### PR TITLE
Make Subs and Derivative classes closer to SymPy

### PR DESCRIPTION
**Pull request by ShikharJ**
Relevant: #119

### CODE REVIEW

**Comment by ShikharJ** [comment]
@isuruf I am not confident that this is how it is supposed to be implemented. Please review and suggest revisions.

**Comment by ShikharJ** [comment]
Ping @isuruf.

**Comment by ShikharJ** [comment]
Ping @isuruf.

File: symengine/lib/symengine_wrapper.pyx [hunk 1]
```diff
@@ -199,9 +199,9 @@ def sympy2symengine(a, raise_error=False):
     elif isinstance(a, sympy.gamma):
         return gamma(a.args[0])
     elif isinstance(a, sympy.Derivative):
-        return Derivative(a.expr, a.variables)
+        return diff(a.expr, a.variables)
     elif isinstance(a, sympy.Subs):
-        return Subs(a.expr, a.variables, a.point)
+        return Subs.subs(a.expr, a.variables, a.point)
```
Comment (isuruf) [inline symengine/lib/symengine_wrapper.pyx]: Make this `a.expr.subs(a.variables, a.point)`

File: symengine/sympy_compat.py [hunk 2]
```diff
@@ -126,6 +126,19 @@ def __new__(cls, a):
...
+
+    def __new__(cls, expr, variables):
+        return symengine.diff(expr, variables)
+
+
+class Subs(_RegisteredFunction):
+    _classes = (symengine.Subs,)
+
+    def __new__(cls, *args):
+        return symengine.Subs.subs(*args)
```
Comment (ShikharJ) [inline symengine/sympy_compat.py]: @isuruf How should this call be modified, as there isn't any standalone `subs` function in the wrapper?

File: symengine/lib/symengine_wrapper.pyx (hunk 1 above)
Comment (isuruf) [inline symengine/lib/symengine_wrapper.pyx]: This won't work either.

Comment (ShikharJ) [inline symengine/lib/symengine_wrapper.pyx]: So, should we first have standalone class of `expr` and function of `subs`?

Comment (ShikharJ) [inline symengine/lib/symengine_wrapper.pyx]: Ping @isuruf.

**Comment by isuruf** [comment]
Replaced by #145

**Pull request closed without merging**